debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/transforms/MongoDataConverter.java

```diff
@@ -316,7 +321,7 @@ public void addFieldSchema(Entry<String, BsonValue> keyValuesforSchema, SchemaBu
                 break;
 
             case ARRAY:
-                if (keyValuesforSchema.getValue().asArray().isEmpty()) {
+                if (keyValuesforSchema.getValue().asArray().isEmpty() && !sanitizeValue) {
```

Comment:
How comes this impacts the _schema_, too? I'd have expected the schema to be the same, no matter whether the value is empty or not?

Reply:
Because if not you will have as schema an Array of String and is not always a array of String. So then when you will try to enforce an avro schema, this schema will try to enforce the real type inside the array.

Reply:
Sorry, still not quite following. The value is an empty array in this case, so how will this fail Avro validation?

Reply:
Ok, I think I got it now. The schema type itself is confusing here. Pushed a commit for some polishing and merged. Thx!